app/Platform/Actions/CopyAchievementUnlocksToEventAchievement.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Models\EventAchievement;
+use App\Models\PlayerAchievement;
+use App\Platform\Jobs\UnlockPlayerAchievementJob;
+
+class CopyAchievementUnlocksToEventAchievement
+{
+    public function execute(
+        EventAchievement $eventAchievement
+    ): void {
+        $eventAchievement->loadMissing(['achievement', 'sourceAchievement']);
+
+        $achievement = $eventAchievement->achievement;
+        $sourceAchievement = $eventAchievement->sourceAchievement;
+
+        if ($achievement && $sourceAchievement) {
+            $achievement->title = $sourceAchievement->title;
+            $achievement->description = $sourceAchievement->description;
+            $achievement->BadgeName = $sourceAchievement->BadgeName;
+            $achievement->save();
```

Comment:
During my local testing and trying to break things, this caught me by surprise. I didn't expect an action named `CopyAchievementUnlocksToEventAchievement` to be mutating these primary data fields on the event achievement.

Maybe it's worth considering a different name for this action.

Reply:
It used to be called `UpdateEventAchievement` and thought that was too generic. I'm open to suggestions.

Reply:
Actually, I think the action can be eliminated and this code just moved into the Observer